certbot-nginx/certbot_nginx/obj.py

```diff
@@ -193,6 +193,11 @@ def __eq__(self, other):
 
         return False
 
+    def __hash__(self):
```

Comment:
I'm not that worried about this, we should almost certainly leave this for now, and I know we do the same thing in Apache to work around the same problems, BUT from the documentation for [\_\_hash\_\_](https://docs.python.org/3/reference/datamodel.html#object.__hash__):

> If a class defines mutable objects and implements an \_\_eq\_\_() method, it should not implement \_\_hash\_\_(), since the implementation of hashable collections requires that a key’s hash value is immutable (if the object’s hash value changes, it will be in the wrong hash bucket).

We don't change vhosts after they're created often, but we do in a few places and this could lead to bugs like the ones Python is warning us about. Might be worth making these objects immutable or writing our own code to eliminate duplicates from a collection of vhosts at some point.

Reply:
https://github.com/certbot/certbot/issues/5647